spark/src/main/java/org/opensearch/sql/spark/functions/response/DefaultSparkSqlFunctionResponseHandle.java

```diff
@@ -64,30 +67,43 @@ private static LinkedHashMap<String, ExprValue> extractRow(
-      } else if (type == ExprCoreType.SHORT) {
-        linkedHashMap.put(column.getName(), new ExprShortValue(row.getInt(column.getName())));
-      } else if (type == ExprCoreType.BYTE) {
-        linkedHashMap.put(column.getName(), new ExprByteValue(row.getInt(column.getName())));
-      } else if (type == ExprCoreType.DOUBLE) {
-        linkedHashMap.put(column.getName(), new ExprDoubleValue(row.getDouble(column.getName())));
-      } else if (type == ExprCoreType.FLOAT) {
-        linkedHashMap.put(column.getName(), new ExprFloatValue(row.getFloat(column.getName())));
-      } else if (type == ExprCoreType.DATE) {
-        // TODO :: correct this to ExprTimestampValue
-        linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
-      } else if (type == ExprCoreType.TIMESTAMP) {
-        linkedHashMap.put(
-            column.getName(), new ExprTimestampValue(row.getString(column.getName())));
-      } else if (type == ExprCoreType.STRING) {
-        linkedHashMap.put(column.getName(), new ExprStringValue(jsonString(row, column.getName())));
+      if (!row.has(column.getName())) {
+        linkedHashMap.put(column.getName(), ExprNullValue.of());
       } else {
-        throw new RuntimeException("Result contains invalid data type");
+        if (type == ExprCoreType.BOOLEAN) {
+          linkedHashMap.put(
+              column.getName(), ExprBooleanValue.of(row.getBoolean(column.getName())));
+        } else if (type == ExprCoreType.LONG) {
+          linkedHashMap.put(column.getName(), new ExprLongValue(row.getLong(column.getName())));
+        } else if (type == ExprCoreType.INTEGER) {
+          linkedHashMap.put(column.getName(), new ExprIntegerValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.SHORT) {
+          linkedHashMap.put(column.getName(), new ExprShortValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.BYTE) {
+          linkedHashMap.put(column.getName(), new ExprByteValue(row.getInt(column.getName())));
+        } else if (type == ExprCoreType.DOUBLE) {
+          linkedHashMap.put(column.getName(), new ExprDoubleValue(row.getDouble(column.getName())));
+        } else if (type == ExprCoreType.FLOAT) {
+          linkedHashMap.put(column.getName(), new ExprFloatValue(row.getFloat(column.getName())));
+        } else if (type == ExprCoreType.DATE) {
+          // TODO :: correct this to ExprTimestampValue
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.TIMESTAMP) {
+          linkedHashMap.put(
+              column.getName(), new ExprTimestampValue(row.getString(column.getName())));
+        } else if (type == ExprCoreType.STRING) {
+          linkedHashMap.put(column.getName(), new ExprStringValue(row.getString(column.getName())));
+        } else {
+          // SparkDataType
+          Object jsonValue = row.get(column.getName());
```

Comment:
I forgot earlier how we added support for non-json response such as Spark EXPLAIN output. This doesn't impact that right?

Reply:
explain result is string value, 

add IT https://github.com/opensearch-project/sql/pull/2492/files#diff-c53435443207a93d99630ec47fde52a91da4e4403b712c9e3bb51aa074b51ce6R335